src/transformations/transform-constructor.ts

```diff
@@ -4,11 +4,36 @@ import { getConstructor, getNodeBounds } from '../solc/ast-utils';
 import { Transformation, TransformHelper } from './type';
 import { buildSuperCallsForChain2 } from './utils/build-super-calls-for-chain';
 import { findAll } from 'solidity-ast/utils';
+import { FunctionDefinition, VariableDeclaration } from 'solidity-ast';
 import { TransformerTools } from '../transform';
 import { newFunctionPosition } from './utils/new-function-position';
 import { formatLines } from './utils/format-lines';
 import { hasConstructorOverride, hasOverride } from '../utils/upgrades-overrides';
 
+// Removes parameters unused by the constructor's body
+function GetUnchainedArguments(constructor: FunctionDefinition, current: string): string {
+  const parameters = constructor.parameters.parameters;
+
+  if (parameters?.length) {
+    const identifiers = [...findAll('Identifier', constructor.body!)];
+    let result: string = current;
+
+    parameters.map((p: VariableDeclaration) => {
+      // Check if parameter is used
+      const found = identifiers.some(id => id.referencedDeclaration === p.id);
+      if (!found) {
+        // Remove unused parameter
+        const reg = new RegExp('\\s' + p.name + '\\,?\\b', 'gi');
+        result = result.replace(reg, '');
```

Comment:
This is not fully robust, because `current` can contain comments (e.g. `foo, bar /* xyz bar */`), also because identifiers can contain `$` which messes up word boundaries (`\b`) and also has a different meaning in regular expressions (match end of string).

I can't think of a way to make it robust against comments, so let's just tackle the other issues.

I believe the best way to do this is to use a regex to iterate over all identifiers including whitespace: `/\s+[a-z0-9$_]+/gi`, and then for each match, replace it with the empty string if it equals the parameter name.

I think this would work:
```suggestion
        result = result.replace(/\s+[a-z0-9$_]+/gi, m => m.trim() === p.name ? '' : m);
```